tap_zendesk/streams.py

```diff
@@ -353,19 +373,57 @@ class TicketAudits(Stream):
     endpoint='https://{}.zendesk.com/api/v2/tickets/{}/audits.json'
     item_key='audits'
 
-    def get_objects(self, ticket_id):
+    async def sync_in_bulk(self, ticket_ids, comments_stream):
+        """
+        Asynchronously fetch ticket audits for multiple tickets
+        """
+        # Create an asynchronous HTTP session
+        async with aiohttp.ClientSession() as session:
+            tasks = [self.sync(session, ticket_id, comments_stream) for ticket_id in ticket_ids]
+            # Run all tasks concurrently and wait for them to complete
+            return await asyncio.gather(*tasks)
+
+    async def get_objects(self, session, ticket_id):
         url = self.endpoint.format(self.config['subdomain'], ticket_id)
-        # Pass `request_timeout` parameter
-        pages = http.get_offset_based(url, self.config['access_token'], self.request_timeout, self.page_size)
-        for page in pages:
-            yield from page[self.item_key]
+        # Fetch the ticket audits using pagination
+        records = await http.paginate_ticket_audits(session, url, self.config['access_token'], self.request_timeout, self.page_size)
 
-    def sync(self, ticket_id):
-        ticket_audits = self.get_objects(ticket_id)
-        for ticket_audit in ticket_audits:
-            zendesk_metrics.capture('ticket_audit')
-            self.count += 1
-            yield (self.stream, ticket_audit)
+        return records[self.item_key]
+
+    async def sync(self, session, ticket_id, comments_stream):
+        """
+        Fetch ticket audits for a single ticket. Also exctract comments for each audit.
+        """
+        audit_records = []
+        comment_records = []
+        try:
+            # Fetch ticket audits for the given ticket ID
+            ticket_audits = await self.get_objects(session, ticket_id)
+            for ticket_audit in ticket_audits:
+                if self.is_selected():
+                    zendesk_metrics.capture('ticket_audit')
+                    self.count += 1
+                    audit_records.append((self.stream, ticket_audit))
+
+                if comments_stream.is_selected():
+                    # Extract comments from ticket audit
+                    ticket_comments = (event for event in ticket_audit['events'] if event['type'] == 'Comment')
+                    zendesk_metrics.capture('ticket_audit')
```

Comment:
Should this be `zendesk_metrics.capture('ticket_audit')`?

Reply:
Yes, it should be `ticket_comments `.